datafusion/src/logical_plan/builder.rs

```diff
@@ -115,21 +115,23 @@ impl LogicalPlanBuilder {
 
     /// Scan a CSV data source
     pub fn scan_csv(
-        path: &str,
+        path: impl Into<String>,
         options: CsvReadOptions,
         projection: Option<Vec<usize>>,
     ) -> Result<Self> {
-        let provider = Arc::new(CsvFile::try_new(path, options)?);
+        let path = path.into();
```

Comment:
As above, this pattern just doesn't make a lot of sense to me -- if someone passed in a `&str` then this code will now create a copy (a `String`) and then simply use a reference to that copy (rather than the path itself)

Reply:
the owned value is passed along to `Self::scan` at the end of the function.